rerun_py/rerun_sdk/rerun/archetypes/points3d.py

```diff
@@ -153,9 +153,91 @@ def _clear(cls) -> Points3D:
+
+        Parameters
+        ----------
+        clear:
+            If true, all unspecified fields will be explicitly cleared.
+        positions:
+            All the 3D positions at which the point cloud shows points.
+        radii:
+            Optional radii for the points, effectively turning them into circles.
+        colors:
+            Optional colors for the points.
+
+            The colors are interpreted as RGB or RGBA in sRGB gamma-space,
+            As either 0-1 floats or 0-255 integers, with separate alpha.
+        labels:
+            Optional text labels for the points.
+
+            If there's a single label present, it will be placed at the center of the entity.
+            Otherwise, each instance will have its own label.
+        show_labels:
+            Optional choice of whether the text labels should be shown by default.
+        class_ids:
+            Optional class Ids for the points.
+
+            The [`components.ClassId`][rerun.components.ClassId] provides colors and labels if not specified explicitly.
+        keypoint_ids:
+            Optional keypoint IDs for the points, identifying them within a class.
+
+            If keypoint IDs are passed in but no [`components.ClassId`][rerun.components.ClassId]s were specified, the [`components.ClassId`][rerun.components.ClassId] will
+            default to 0.
+            This is useful to identify points within a single classification (which is identified
+            with `class_id`).
+            E.g. the classification might be 'Person' and the keypoints refer to joints on a
+            detected skeleton.
+
+        """
+
+        kwargs = {
+            "positions": positions,
+            "radii": radii,
+            "colors": colors,
+            "labels": labels,
+            "show_labels": show_labels,
+            "class_ids": class_ids,
+            "keypoint_ids": keypoint_ids,
+        }
+
+        if clear:
+            kwargs = {k: v if v is not None else [] for k, v in kwargs.items()}  # type: ignore[misc]
+
+        return Points3D(**kwargs)  # type: ignore[arg-type]
+
+    @classmethod
+    def clear_fields(cls) -> Points3D:
+        """Clear all the fields of a `Points3D`."""
+        inst = cls.__new__(cls)
+        inst.__attrs_init__(
+            positions=[],  # type: ignore[arg-type]
+            radii=[],  # type: ignore[arg-type]
+            colors=[],  # type: ignore[arg-type]
+            labels=[],  # type: ignore[arg-type]
+            show_labels=[],  # type: ignore[arg-type]
+            class_ids=[],  # type: ignore[arg-type]
+            keypoint_ids=[],  # type: ignore[arg-type]
+        )
+        return inst
+
     positions: components.Position3DBatch = field(
-        metadata={"component": "required"},
-        converter=components.Position3DBatch._required,  # type: ignore[misc]
+        metadata={"component": "optional"},
+        converter=components.Position3DBatch._optional,  # type: ignore[misc]
```

Comment:
Typing wise, all component become optional with my proposal. So we can remove `BaseBatch._required` and rename `BaseBatch._optional` to `_converter` or something.